.buildkite/pipeline.py

```diff
@@ -0,0 +1,149 @@
+    SupportedPython.V2_7: "python:2.7",
+}
+
+TOX_MAP = {
+    SupportedPython.V3_7: "py37",
+    SupportedPython.V3_6: "py36",
+    SupportedPython.V3_5: "py35",
+    SupportedPython.V2_7: "py27",
+}
+
+
+class StepBuilder:
+    def __init__(self, label):
+        self._step = {"label": label}
+
+    def run(self, *argc):
+        self._step["command"] = "\n".join(argc)
+        return self
+
+    def onImage(self, img):
+        if img in IMAGE_MAP:
+            img = IMAGE_MAP[img]
+        self._step["plugins"] = [{DOCKER_PLUGIN: {"image": img}}]
+        return self
+
+    def withEnvironVar(self, var):
+        self._step["env"] = {var: "${{{}?}}".format(var)}
+        return self
+
+    def build(self):
+        return self._step
+
+
+def wait_step():
+    return "wait"
+
+
+def python_modules_tox_tests(directory, prereqs=None):
+    label = directory.replace("/", "-")
+    tests = []
+    for version in SupportedPythons:
+        coverage = ".coverage.{}.{}.$BUILDKITE_BUILD_ID".format(label, version)
+        tox_command = []
+        if prereqs:
+            tox_command += prereqs
+        tox_command += [
+            "pip install tox;",
+            "cd python_modules/{}".format(directory),
+            "tox -e {}".format(TOX_MAP[version]),
+            "mv .coverage {}".format(coverage),
+            "buildkite-agent artifact upload {}".format(coverage),
+        ]
+        tests.append(
+            StepBuilder("{} tests ({})".format(directory, version))
+            .run(*tox_command)
+            .onImage(version)
+            .build()
+        )
+
+    return tests
+
+
+if __name__ == "__main__":
+    steps = [
+        StepBuilder("pylint")
+        .run("make dev_install", "make pylint")
+        .onImage(SupportedPython.V3_7)
+        .build(),
+        StepBuilder("black")
+        .run("pip install black==18.9b0", "make check_black")
+        .onImage(SupportedPython.V3_7)
+        .build(),
+        StepBuilder("docs snapshot test")
+        .run(
+            "pip install -r python_modules/dagster/dev-requirements.txt -qqq",
+            "pip install -e python_modules/dagster -qqq",
+            "pytest -vv python_modules/dagster/docs",
+        )
+        .onImage(SupportedPython.V3_7)
+        .build(),
+        StepBuilder("dagit webapp tests")
+        .run(
+            "pip install -r python_modules/dagster/dev-requirements.txt -qqq",
+            "pip install -e python_modules/dagster -qqq",
+            "pip install -e python_modules/dagster-graphql -qqq",
+            "pip install -e python_modules/dagit -qqq",
+            "pip install -r python_modules/dagit/dev-requirements.txt -qqq",
+            "cd js_modules/dagit",
+            "yarn install --offline",
+            "yarn run ts",
+            "yarn run jest",
+            "yarn run check-prettier",
+            "yarn generate-types",
+            "git diff --exit-code",
+        )
+        .onImage("nikolaik/python-nodejs:python3.7-nodejs11")
```

Comment:
yiiikes

Reply:
ill sync up with you tomorrow for what to do here, maybe our own image